plugins/modules/mysql_db.py

```diff
@@ -422,13 +430,20 @@ def db_dump(module, host, user, password, db_name, target, all_databases, port,
     elif os.path.splitext(target)[-1] == '.xz':
         path = module.get_bin_path('xz', True)
 
-    if path:
+    if pipefail:
+        cmd = 'set -o pipefail && %s | %s > %s' % (cmd, path, shlex_quote(target))
+    elif path:
```

Comment:
```suggestion
    if pipefail:
        cmd = 'set -o pipefail && %s | %s > %s' % (cmd, path, shlex_quote(target))
    elif path:
```
are they mutually exclusive with `path`?

Reply:
uh, you're right, I should have written:
```python
if pipefail and path:
  cmd = '...'
elif path:
  cmd = '...'
 else:
   cmd += '...'
```